eng/nuget-client-versions.json

```diff
@@ -0,0 +1,10 @@
+[
+  {
+    "filter": "4.4.*",
```

Comment:
I was expecting a pattern more like the lcu-config where the Fx versions were specified to determine the behavior rather than a wild card pattern.  I thought this because of my experiences with .NET Core.  When new .NET versions come out, you can easily copy the previous Dockerfiles and use the updater tool to modify the copied Dockerfiles.  With this pattern both the current and new Dockerfiles would get updated.  In reality this mater here since 4.8 is the last .NET FX version.

I think the `filter` property could be given a more descriptive name to make it more meaningful.  Maybe MatchVersionFilter?

Reply:
Except it's not going to vary by framework version; it'll vary by OS version.  Would you be ok with something like this:

```JSON
[
  {
    "nugetClientVersion": "4.4.3",
    "osVersions": [
      "windowsservercore-1903",
      "windowsservercore-ltsc2016",
      "windowsservercore-ltsc2019"
    ]
  },
  {
    "nugetClientVersion": "5.3.1",
    "osVersions": [
      "windowsservercore-1909"
    ]
  }
]
```

Reply:
Good point on the OS version.  Yes I like this format.